megamek/src/megamek/common/Entity.java

```diff
@@ -2459,34 +2514,41 @@ public String getDisplayName() {
         return displayName;
     }
 
+    /**
+     * Sets the display name for this entity.
+     * @param displayName The new display name.
+     */
+    protected void setDisplayName(String displayName) {
+        this.displayName = displayName;
+    }
+
     /**
      * Generates the display name for this entity.
      * <p/>
      * Sub-classes are allowed to override this method. The display name is in
      * the format [Chassis] [Model] ([Player Name]).
      */
     public void generateDisplayName() {
-        StringBuffer nbuf = new StringBuffer();
-        nbuf.append(chassis);
-        if ((model != null) && (model.length() > 0)) {
-            nbuf.append(" ").append(model);
-        }
-        // if show unit id is on, append the id
-        if (PreferenceManager.getClientPreferences().getShowUnitId()) {
-            nbuf.append(" ID:").append(getId());
-        } else if (duplicateMarker > 1) {
-            // if not, and a player has more than one unit with the same name,
-            // append "#N" after the model to differentiate.
-            nbuf.append(" #" + duplicateMarker);
+        synchronized (nameLock) {
+            displayName = createDisplayName(duplicateMarker);
         }
+     }
+
+    /**
+     * Creates a display name for the entity.
+     * @param duplicateMarker A number used to disambiguate two entities with
+     *                        the same name.
+     * @return A display name for the entity.
+     */
+    private String createDisplayName(int duplicateMarker) {
+        StringBuilder nbuf = new StringBuilder();
```

Comment:
nbuf is a strange name for a StringBuilder

Reply:
I left the original names in the refactor.